bwc-test/src/test/java/org/opensearch/security/bwc/SecurityBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,367 @@
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertPluginUpgrade(String uri) throws Exception {
+        Map<String, Map<String, Object>> responseMap = (Map<String, Map<String, Object>>) getAsMap(uri).get("nodes");
+        for (Map<String, Object> response : responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
+            Set<String> pluginNames = plugins.stream().map(map -> (String) map.get("name")).collect(Collectors.toSet());
+
+            final Version minNodeVersion = minimumNodeVersion();
+
+            if (minNodeVersion.major <= 1) {
+                assertThat(pluginNames, hasItem("opensearch_security")); // With underscore seperator
+            } else {
+                assertThat(pluginNames, hasItem("opensearch-security")); // With dash seperator
+            }
+        }
+    }
+
+    /**
+     * Ingests data into the test index
+     * @param index index to ingest data into
+     */
+
+    private void ingestData(String index) throws IOException {
+        StringBuilder bulkRequestBody = new StringBuilder();
+        ObjectMapper objectMapper = new ObjectMapper();
+        int numberOfRequests = Randomness.get().nextInt(10);
+        while (numberOfRequests-- > 0) {
+            for (int i = 0; i < Randomness.get().nextInt(100); i++) {
+                Map<String, Map<String, String>> indexRequest = new HashMap<>();
+                indexRequest.put("index", new HashMap<>() {
+                    {
+                        put("_index", index);
+                    }
+                });
+                bulkRequestBody.append(objectMapper.writeValueAsString(indexRequest) + "\n");
+                bulkRequestBody.append(objectMapper.writeValueAsString(Song.randomSong().asJson()) + "\n");
+            }
+            List<Response> responses = RestHelper.requestAgainstAllNodes(
+                testUserRestClient,
+                "POST",
+                "_bulk?refresh=wait_for",
+                RestHelper.toHttpEntity(bulkRequestBody.toString())
+            );
+            responses.forEach(r -> assertEquals(200, r.getStatusLine().getStatusCode()));
+        }
+    }
+
+    /**
+     * Runs a matchAll query against the test index
+     * @param index index to search
+     */
+    private void searchMatchAll(String index) throws IOException {
+        String matchAllQuery = "{\n" + "    \"query\": {\n" + "        \"match_all\": {}\n" + "    }\n" + "}";
+        int numberOfRequests = Randomness.get().nextInt(10);
+        while (numberOfRequests-- > 0) {
+            List<Response> responses = RestHelper.requestAgainstAllNodes(
+                testUserRestClient,
+                "POST",
+                index + "/_search",
+                RestHelper.toHttpEntity(matchAllQuery)
+            );
+            responses.forEach(r -> assertEquals(200, r.getStatusLine().getStatusCode()));
+        }
```

Comment:
Had added multiple requests per test as we had earlier observed non-deterministic behaviour owing to how OS may route requests to various nodes or shards. The match_all query behaviour is random and it can query from either primaries or replicas.

Reply:
Do you know if we ensure this kind of coverage by using `routing` or `preference` parameters [1]?

- [1] https://opensearch.org/docs/latest/api-reference/search/#the-preference-query-parameter